dbt/include/dremio/macros/adapters/apply_grants.sql

```
@@ -22,18 +22,25 @@ limitations under the License.*/
     {%- else -%}
         {%- set relation_without_double_quotes = relation.database ~ '.' ~ relation.schema ~ '.' ~ relation.identifier-%}
     {%- endif %}
-
+    
+    {%- if target.cloud_host and not target.software_host -%}
+        {%- set privileges_table = 'sys.project.privileges' -%}
+    {%- elif target.software_host and not target.cloud_host -%}
+        {%- set privileges_table = 'sys.privileges' -%}
+    {%- else -%}
+        {%- set privileges_table = 'sys.privileges' -%}
```

Can we throw an error here?

---

Good suggestion. Done.